tests/test_not_finite.py

```diff
@@ -0,0 +1,249 @@
+"""
+A collection of tests related to non-finite values in shaders, like nan and inf.
+
+See:
+* https://en.wikipedia.org/wiki/NaN
+* https://github.com/gpuweb/gpuweb/pull/2311#issuecomment-1973533433
+
+"""
+
+import ctypes
+
+import numpy as np
+
+from wgpu.utils.compute import compute_with_buffers
+from pytest import skip
+from testutils import can_use_wgpu_lib
+
+
+if not can_use_wgpu_lib:
+    skip("Skipping tests that need the wgpu lib", allow_module_level=True)
+
+
+def test_finite_using_nequal():
+    # Just to demonstrate that this does not work.
+    # The compiler filters optimizes away the check.
+
+    shader = """
+        @group(0)
+        @binding(0)
+        var<storage,read> values: array<f32>;
+
+        fn is_nan(v:f32) -> bool {
+            return v != v;
+        }
+
+        fn is_inf(v:f32) -> bool {
+            return v != 0.0 && v * 2.0 == v;
+        }
+
+        fn is_finite(v:f32) -> bool {
+            return v == v && v * 2.0 != v;
+        }
+
+        fn to_real(v:f32) -> f32 {
+            return select(0.0, v, is_finite(v));
+        }
+
+    """
+
+    detect_finites("nequal", shader, False, False)
+
+
+def test_finite_using_min_max():
+    # This obfuscates the check for equality enough for the compiler
+    # not to optimize it away.
+    #
+    # However, if fastmath is enabled, depending on the hardare/compiler,
+    # the loaded value may not actually be a nan/inf anymore.
```

Comment:
So we were unable to see this on the various platforms you tried right?

Reply:
Yes, everything is fine with the min-max approach on all platforms I tested this on.